src/explore-education-statistics-admin/src/pages/release/edit-release/content/actions.ts

```diff
@@ -0,0 +1,90 @@
+import { EditableContentBlock } from '@admin/services/publicationService';
+import {
+  AbstractRelease,
+  ContentSection,
+} from '@common/services/publicationService';
+import { State } from './ReleaseContext';
+
+type ClearState = { type: 'CLEAR_STATE' };
+type SetState = { type: 'SET_STATE'; payload: State };
+type SetAvailableDatablocks = {
+  type: 'SET_AVAILABLE_DATABLOCKS';
+  payload: Pick<State, 'availableDataBlocks'>;
+};
+export type RemoveBlockFromSection = {
+  type: 'REMOVE_BLOCK_FROM_SECTION';
+  payload: {
+    meta: {
+      sectionId: string;
+      blockId: string;
+      sectionKey: keyof Pick<
+        AbstractRelease<EditableContentBlock>,
+        | 'summarySection'
+        | 'keyStatisticsSection'
+        | 'keyStatisticsSecondarySection'
+        | 'headlinesSection'
+        | 'content'
+      >;
+    };
+  };
+};
+export type UpdateBlockFromSection = {
+  type: 'UPDATE_BLOCK_FROM_SECTION';
+  payload: {
+    block: EditableContentBlock;
+    meta: RemoveBlockFromSection['payload']['meta'];
+  };
+};
+
+export type AddBlockToSection = {
+  type: 'ADD_BLOCK_TO_SECTION';
+  payload: {
+    block: EditableContentBlock;
+    meta: Omit<RemoveBlockFromSection['payload']['meta'], 'blockId'>;
```

Comment:
Why are we even omitting `blockId`?

Reply:
We're adding a new block, not updating a specific block.
We don't care about the id of the block, just the id of the section it's going to be placed in here

Reply:
I wonder if update/add actions could be merged and then the only difference in behaviour is if the `blockId` is set or not (i.e. make it optional).

I know that the meta type definitely needs to make `blockId` optional at least.

Reply:
I prefer to keep them separate, I think ADD and UPDATE actions mirror the POST and PUT calls or the CREATE and UPDATE of crud